### PR TITLE
[WIP]: upgrade ember-cli to 2.4

### PR DESCRIPTION
**Pull request by snewcomer**
## Here is a WIP to upgrade ember-cli to 2.4

- obj[key] to set a key, however, need to use Ember.set.  One avenue seems to be to also upgrade ember as well as I left it at 1.13.

Failing test - https://github.com/hhff/ember-infinity/blob/master/tests/acceptance/infinity-route-with-embedded-route-test.js#L19

screen shot 2017-04-06 at 4 34 52 pm

### CODE REVIEW

File: tests/helpers/start-app.js [hunk 1]
```diff
@@ -2,13 +2,11 @@ import Ember from 'ember';
...
 
-const assign = Ember.assign || Ember.merge;
-
 export default function startApp(attrs) {
   let application;
 
-  let attributes = assign({}, config.APP);
-  attributes = assign(attributes, attrs); // use defaults, but you can override;
+  let attributes = Ember.merge({}, config.APP);
+  attributes = Ember.merge(attributes, attrs); // use defaults, but you can override;
```
Comment (hhff) [inline tests/helpers/start-app.js]: Why is this change necessary?  I thought merge was deprecated for assign

Comment (snewcomer) [inline tests/helpers/start-app.js]: This just came from running ember init on 2.4

Comment (hhff) [inline tests/helpers/start-app.js]: ohh right sorry read this wrong!

**Comment by hhff** [comment]
Looks to be a bunch of failing tests on Travis.  Are you seeing them locally?

**Comment by snewcomer** [comment]
@hhff yeah seeing them locally as well.  Some have to do with backburner and the likes.  This PR was mainly to show trying to upgrade ember-cli to 2.4 and leaving ember where it is at.  You mentioned support for 1.10.  I wonder if to get the tests passing, we would need to take an approach like [this](https://github.com/cibernox/ember-power-select/blob/master/config/ember-try.js).  I haven't been around the ember community long enough to understand how to maintain backwards compat and get all of these tests passing.

**Comment by hhff** [comment]
@snewcomer im cool with switching to the ember-power-select way and dropping 1.x!  Let's do that!

File: tests/unit/mixins/route-test.js [hunk 2]
```diff
@@ -87,13 +85,12 @@ test('it can not use infinityModel without a Model Name', assert => {
       query() {}
     }
   });
-
-  assert.throws(() => {
+ 
+  try {
```
Comment (snewcomer) [inline tests/unit/mixins/route-test.js]: assert.throws failes after ember 2.11 so wrap try catch.  However, this method does not work in the infinity-loader unit test.

File: tests/unit/components/infinity-loader-test.js [hunk 3]
```diff
@@ -88,7 +89,7 @@ test('it uses the provided scrollable element', function(assert) {
   assert.equal(scrollable[0], $("#content")[0]);
 });
 
-test('it throws error when scrollable element is not found', function(assert) {
+skip('it throws error when scrollable element is not found', function(assert) {
   assert.expect(1);
 
   this.subject({scrollable: "#nonexistent"});
```
Comment (snewcomer) [inline tests/unit/components/infinity-loader-test.js]: wrapping try catch around this only passes after reload of test.  Fails on first run though.  So assuming state issue.  Need to dig further.

File: tests/unit/mixins/route-test.js (hunk 2 above)
Comment (snewcomer) [inline tests/unit/mixins/route-test.js]: https://github.com/emberjs/ember.js/pull/14898

**Comment by snewcomer** [comment]
@hhff yayyy!  Here is a passing build tests against 2.4 and above.  Used firefox to run the tests on travis as well.  Let me know what you think.

File: tests/unit/mixins/route-test.js [hunk 4]
```diff
@@ -55,12 +55,11 @@ function callModelHook(route) {
 test('it can not use infinityModel without Ember Data Store', assert => {
   var route = createRoute(['post'], {store: null});
 
-  assert.throws(() => {
+  try {
     route.model();
-  },
```
Comment (snewcomer) [inline tests/unit/mixins/route-test.js]: wrapping with try catch seems to work when catching this error.

**Comment by snewcomer** [comment]
I'm guessing to get these other tests to pass I need to upgrade other deps including ember-qunit and potentially changing the acceptance tests to use `moduleForAcceptance`.  Any thoughts on using mirage or would you like to keep the fake server?

File: tests/unit/components/infinity-loader-test.js (hunk 3 above)
Comment (snewcomer) [inline tests/unit/components/infinity-loader-test.js]: Have tried many solutions [here](https://github.com/emberjs/ember.js/pull/14898) except ember-qunit-assert-helpers sol'n.  I need to upgrade the test suite to be able to use this.

File: tests/unit/mixins/route-test.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
 import Ember from 'ember';
+import DS from 'ember-data';
 import RouteMixin from 'ember-infinity/mixins/route';
 import { module, test } from 'qunit';
 
 module('RouteMixin');
 
 const assign = Ember.assign || Ember.merge;
 
-let EO = Ember.Object.create.bind(Ember.Object);
```
Comment (snewcomer) [inline tests/unit/mixins/route-test.js]: Doing this failed on test runs.

Comment (hhff) [inline tests/unit/mixins/route-test.js]: Im fine with this!

File: tests/unit/mixins/route-test.js (hunk 2 above)
Comment (hhff) [inline tests/unit/mixins/route-test.js]: kk

File: tests/unit/components/infinity-loader-test.js (hunk 3 above)
Comment (hhff) [inline tests/unit/components/infinity-loader-test.js]: Technically this error throws in didInsertElement.

Could we try something more specific to that method?

```
try {
  this.didInsertElement();
} catch
```
or something like that?

Comment (snewcomer) [inline tests/unit/components/infinity-loader-test.js]: Brilliant!  Yep that worked. pushed
